packages/react/src/floating-ui-react/hooks/useListNavigation.ts

```diff
@@ -451,8 +453,6 @@ export function useListNavigation(
           }
         };
 
-        forceSyncFocusRef.current = false;
```

Comment:
This assignment was moved down to improve highlight hover perf, but it causes the mentioned bug. So instead, we can make the item mouse listeners be responsible for making the focus sync, which retains the perf improvement while addressing this issue. They still guard `focusItemOnHover`, and only let the refs be assigned